index.js

```diff
@@ -10,6 +10,12 @@ async function fastifyElasticsearch (fastify, options) {
 
   const client = options.client || new Client(options)
 
+  if (!fastify.isElasticsearchClient) {
+    fastify.decorate('isElasticsearchClient', function isElasticsearchClient (value) {
```

Comment:
Why is this needed?

Reply:
If the question is why it is decorated to the fastify.instance:
The reason, why I added that function/typeguard was, that I did not wanted to enforce the need to require the plugin again just to import isElasticsearchClient to use it for typescript purposes. 
Also the fastify-instance scope felt correct, as the elastic search instance is imho connected on fastify start. 

If the questoin is, why do we have isElasticsearchClient:
In Typescript you need to narrow down the type as the type is actually ambigous. And with isElasticseachClient we can use it to determine if we use namespaces or not. So this is more like a typescript helper method.

Reply:
If the question is why it is decorated to the fastify.instance:
The reason, why I added that function/typeguard was, that I did not wanted to enforce the need to require the plugin again just to import isElasticsearchClient to use it for typescript purposes. 
Also the fastify-instance scope felt correct, as the elastic search instance is imho connected on fastify start. 

If the questoin is, why do we have isElasticsearchClient:
In Typescript you need to narrow down the type as the type is actually ambigous. And with isElasticseachClient we can use it to determine if we use namespaces or not. So this is more like a typescript helper method.